core-bundle/src/EventListener/PurgeExpiredMemberRegistrationsListener.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\MemberModel;
+
+/**
+ * Purges the member registrations in the front and back end, whenever tl_member is loaded (#3711).
+ *
+ * @internal
+ *
+ * @Hook("reviseTable")
+ */
+class PurgeExpiredMemberRegistrationsListener
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(ContaoFramework $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function __invoke(string $table): ?bool
```

Comment:
IIRC, the method always returns `null`, so the return type does not match the method body. Which one is wrong?

Reply:
The code that executes the hook checks the return type for `=== true`. So any hook implementation can either return `true` or something arbitrary, that's why I chose this return type. We can also define it as `void` as it would not matter in this case (just feels wrong to me).